File: cap8/exemplo3/stack.test.js
/**
* Engenharia de Software Moderna - Testes  (Cap. 8)
* Prof. Marco Tulio Valente
* 
* Exemplo de Teste de Unidade - Classe Pilha (em JavaScript)
*
*/

const stack = require('./stack');

describe('teste da classe pilha', () => {

beforeEach(() => {
  s = new stack();
})

test('pilha vazia', () => {
  expect(s.isEmpty()).toBeTruthy();;
});

test('pilha não vazia', () => {
  s.push(5);
  expect(s.isEmpty()).toBeFalsy();
});

test('pilha com 3 elementos', () => {
  s.push(10);
  s.push(20);
  s.push(30);
  expect(s.size()).toBe(3);
});

test('pilha após desempilhar dois elementos', () => {
  s.push(10);
  s.push(20);
  s.push(30);
  s.pop()
  expect(s.pop()).toBe(20);
});

test('pilha com exceção de pilha vazia ao desempilhar', () => {
  s.push(10);
  s.pop();
  expect(() => s.pop()).toThrow();
});

})

File: cap8/exemplo1/src/test/java/exemplo1/StackTest.java
/**
* Engenharia de Software Moderna - Testes  (Cap. 8)
* Prof. Marco Tulio Valente
* 
* Exemplo de Teste de Unidade - Classe Stack
*
*/

package exemplo1;

import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.BeforeEach;

import static org.junit.jupiter.api.Assertions.assertTrue;
import static org.junit.jupiter.api.Assertions.assertFalse;
import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertThrows;

public class StackTest {
  Stack<Integer> stack;

  @BeforeEach
  public void init() {
	stack = new Stack<Integer>();
  }

  @Test
  public void testEmptyStack() {
	assertTrue(stack.isEmpty());
  }

  @Test
  public void testNotEmptyStack() {
	stack.push(10);
	assertFalse(stack.isEmpty());
  }

  @Test
  public void testSizeStack() {
	stack.push(10);
	stack.push(20);
	stack.push(30);
	int size = stack.size();
	assertEquals(3, size);
  }

  @Test
  public void testPushPopStack() {
	stack.push(10);
	stack.push(20);
	stack.push(30);
	int result = stack.pop();
	result = stack.pop();
	assertEquals(20, result);
  }

	
  @Test
  public void testEmptyStackException() {
	stack.push(10);
	stack.pop();
	assertThrows(java.util.EmptyStackException.class, () -> {
	  stack.pop();
    });
  }

}